client/state/plugins/installed/selectors.js

```diff
@@ -87,7 +87,7 @@ export function getPlugins( state, siteIds, pluginFilter ) {
 export function getPluginsWithUpdates( state, siteIds ) {
 	return filter( getPlugins( state, siteIds ), _filters.updates ).map( plugin => ( {
 		...plugin,
-		version: plugin.update.new_version,
+		version: plugin.update && plugin.update.new_version,
```

Comment:
This looks good, but perhaps we could use the new optional chaining here:

```suggestion
		version: plugin?.update?.new_version,
```